litex/soc/integration/builder.py

```diff
@@ -187,7 +187,7 @@ def _generate_rom_software(self, compile_bios=True):
                 dst_dir = os.path.join(self.software_dir, name)
                 makefile = os.path.join(src_dir, "Makefile")
                 if self.compile_software:
-                    subprocess.check_call(["make", "-C", dst_dir, "-f", makefile])
+                    subprocess.check_call(["make", "-j", "4", "-C", dst_dir, "-f", makefile])
```

Comment:
I wonder if this should be something like `os.environ.get('MAKEFLAGS', '')`? See https://www.gnu.org/software/make/manual/html_node/Options_002fRecursion.html

Reply:
Also https://www.gnu.org/software/make/manual/html_node/Parallel.html#Parallel

Reply:
I ended up proposing this change because MAKEFLAGS was apparently not being respected.

On a closer reading of the documentation, that's because litex is being started in a recursively executed makefile, and the outermost makefile is likely stripping -j from MAKEFLAGS. 

Maybe the better approach is to ensure that the MAKEFLAGS variable contains a -j flag.

Reply:
Is the other Makefile telling make it needs to give the LiteX command the flags (via prefixing the command with a `+`)? https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html#MAKE-Variable and https://www.gnu.org/software/make/manual/html_node/Job-Slots.html#Job-Slots

Reply:
PS Anyone ever tried http://bashdb.sourceforge.net/remake/ ?

Reply:
All Makefiles calling "make" should call "$(MAKE)" instead.